templates/server.properties.erb

```diff
@@ -2,8 +2,8 @@
 #
 # See: http://kafka.apache.org/documentation.html#brokerconfigs for more details
 <%- @server_config.sort.map do |k,v|
-  if v.to_s.strip.length != 0 %>
+    if v.to_s.strip.length != 0 -%>
 <%= k %>=<%= v %>
```

Comment:
Unfortunately this syntax also requires a second end, but your solutions is looking much cleaner.